libr/io/io_cache.c

```diff
@@ -21,37 +21,104 @@ static void cache_item_free(RIOCache *cache) {
 	free (cache);
 }
 
-R_API bool r_io_cache_at(RIO *io, ut64 addr) {
-	RIOCache *c;
-	RListIter *iter;
-	r_list_foreach (io->cache, iter, c) {
-		if (r_itv_contain (c->itv, addr)) {
-			return true;
+#define CMP_END_GTE_PART(addr, part) \
+	(((addr) < (r_itv_end (((RIOCacheSkyline *)(part))->itv)) || !r_itv_end (((RIOCacheSkyline *)(part))->itv)) ? -1 : 1)
+
+#define CMP_BEGIN_GTE_PART(addr, part) \
+	(((addr) > (r_itv_begin (((RIOCacheSkyline *)(part))->itv))) - ((addr) < (r_itv_begin (((RIOCacheSkyline *)(part))->itv))))
+
+// This is copy-pasta from io_map.c
+static bool add_cache_to_skyline(RIO *io, RIOCache *cache) {
```

Comment:
Can we have this skyline thing generic to avoid copypasta?

Reply:
@trufae created RSkyline